spyder_line_profiler_5/spyder/plugin.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+# ----------------------------------------------------------------------------
+# Copyright © 2021, Spyder Line Profiler 5
+#
+# Licensed under the terms of the MIT license
+# ----------------------------------------------------------------------------
+"""
+Spyder Line Profiler 5 Plugin.
+"""
+
+# Third-party imports
+from qtpy.QtGui import QIcon
+
+# Spyder imports
+from spyder.api.plugins import Plugins, SpyderDockablePlugin
+from spyder.api.translations import get_translation
+from spyder.api.plugin_registration.decorators import on_plugin_available
+
+# Local imports
+from spyder_line_profiler_5.spyder.confpage import SpyderLineProfiler5ConfigPage
+from spyder_line_profiler_5.spyder.widgets import SpyderLineProfiler5Widget
+
+_ = get_translation("spyder_line_profiler_5.spyder")
+
+
+class SpyderLineProfiler5(SpyderDockablePlugin):
+    """
+    Spyder Line Profiler 5 plugin.
+    """
+
+    NAME = "spyder_line_profiler_5"
+    REQUIRES = [Plugins.Editor]
+    OPTIONAL = []
+    TABIFY = Plugins.Help
+    WIDGET_CLASS = SpyderLineProfiler5Widget
+    CONF_SECTION = NAME
+    CONF_WIDGET_CLASS = SpyderLineProfiler5ConfigPage
+
+    # --- Signals
+
+    # --- SpyderDockablePlugin API
+    # ------------------------------------------------------------------------
+    def get_name(self):
```

Comment:
I've just confirmed with a spyder developer that for compatibility with 5.2.0+, this has to be turned into a `@staticmethod`.

Reply:
Ok thanks, I integrated that.

However, in the documentation this is not mentioned like that https://docs.spyder-ide.org/current/workshops/plugin-development.html#the-plugin-structure

Reply:
Is this supposedly the same for the other functions i.e. `get_description`, `get_icon` etc?

Reply:
You would think, but according to this comment this is the only change to the API: https://github.com/spyder-ide/spyder/issues/16966#issuecomment-996079320

Reply:
> Before I change it: Is this the same for the other functions i.e. get_description, get_icon etc?

No, it's only necessary for `get_name`.